### PR TITLE
Fixing close icon behaviour for search fields

### PR DESCRIPTION
**Pull request by e34lin**
Addresses the issues on https://ucsc-cgl.atlassian.net/browse/SEAB-2846

Notes:
- Not sure why the close icon on the home page doesn't align with the text, so I just added inline styling, not sure if thats the best solution
- Not able to reproduce "Search Organizations" cutting off when there is an input

How it currently looks like:

Screen Shot 2021-07-12 at 1 40 25 PM
Screen Shot 2021-07-12 at 1 47 23 PM

### CODE REVIEW

**Review by garyluu — COMMENTED** [review]
I'm going to spend a bit of time to experiment with fixes for this

**Comment by e34lin** [comment]
> I'm going to spend a bit of time to experiment with fixes for this

okay, thank you!

**Review by denis-yuen — COMMENTED** [review]
re-request review when appropriate

**Review by coverbeck — COMMENTED** [review]
Same, going to wait for Gary's comments, please re-request review.

File: src/app/organizations/organizations/organizations.component.html [hunk 1]
```diff
@@ -44,9 +44,9 @@
         </mat-form-field>
         <mat-form-field appearance="outline" class="search-form-field">
           <mat-label>Search Organizations</mat-label>
-          <input #search matInput formControlName="name" />
+          <input #search matInput formControlName="name" [(ngModel)]="search.value" />
           <button mat-button *ngIf="search.value" matSuffix mat-icon-button (click)="search.value = ''">
-            <mat-icon inline="true">close</mat-icon>
+            <mat-icon style="font-size: 16px">close</mat-icon>
```
Comment (garyluu) [inline src/app/organizations/organizations/organizations.component.html:49]: inline style https://github.com/dockstore/dockstore/wiki/Dockstore-Frontend-Opinionated-Style-Guide#pr-checklist

**Review by denis-yuen — COMMENTED** [review]
test failure does seem related to changes

File: src/app/organizations/organizations/organizations.component.html (hunk 1 above)
Comment (e34lin) [reply src/app/organizations/organizations/organizations.component.html]: Actually changed the size back to the original to match the sizes of the search page and home page, but could take suggestions and opinions if the smaller close icon looks better on the organization page.

**Review by coverbeck — COMMENTED** [review]
Was here, waiting for tests to pass.

File: src/app/search/basic-search/basic-search.component.html [hunk 2]
```diff
@@ -12,6 +12,7 @@
       list="suggestions"
       type="text"
       data-cy="basic-search"
+      [(ngModel)]="search.value"
```
Comment (garyluu) [inline src/app/search/basic-search/basic-search.component.html:15]: what's the purpose of these 2 ngModel??

Comment (e34lin) [reply src/app/search/basic-search/basic-search.component.html:15]: I found that it fixes the issue addressed https://ucsc-cgl.atlassian.net/browse/SEAB-2846, so when you click on the close icon the search results are cleared. I am not sure if there is another way in fixing this.

Comment (garyluu) [reply src/app/search/basic-search/basic-search.component.html:15]: the oddity is that this component is using formControl, `#search`, and also ngModel. Try to remove one at least

**Review by coverbeck — COMMENTED** [review]
I'm going to wait until Gary's latest comments are addressed. Please request review again when done. Thanks.

**Review by garyluu — APPROVED** [review]
weird merge with the JAR changes, but it's probably fine?

Didn't check this out locally, I'm just assuming everything works.

**Review by denis-yuen — APPROVED** [review]

**Review by kathy-t — APPROVED** [review]

**Review by coverbeck — APPROVED** [review]

**Pull request merged**